extensions/wikia/FilePage/FilePageHelper.class.php

```diff
@@ -13,4 +13,90 @@ public static function stripCategoriesFromDescription( $content ) {
 
 		return $content;
 	}
+
+	/**
+	 * Returns an url when file needs to be redirected.
+	 *
+	 * @requestParam Title
+	 *
+	 * @return string $url - url to redirect to
+	 */
+	public static function fileRedir(ImagePage $page) {
+		global $wgMemc;
+
+		$page = $page->getContext();
```

Comment:
Rather than overwriting `$page` I'd suggest making this `$context`. Though, other than the hook (see my question below), this is only used to get the title, so can this either be `$title = $page->getContext()->getTitle()`, or probably better is to just have the method take the title as the parameter, rather than the Page object?